src/openms/include/OpenMS/CHEMISTRY/ModificationDefinitionsSet.h

```diff
@@ -133,22 +133,37 @@ namespace OpenMS
     */
     //@{
     /// assignment operator
-    ModificationDefinitionsSet & operator=(const ModificationDefinitionsSet & element);
+    ModificationDefinitionsSet& operator=(const ModificationDefinitionsSet& element);
     //@}
 
     /** @name Predicates
     */
     //@{
     /// returns true if the peptide is compatible with the definitions, e.g. does not contain other modifications
-    bool isCompatible(const AASequence & peptide) const;
+    bool isCompatible(const AASequence& peptide) const;
 
     /// equality operator
-    bool operator==(const ModificationDefinitionsSet & rhs) const;
+    bool operator==(const ModificationDefinitionsSet& rhs) const;
 
     /// inequality operator
-    bool operator!=(const ModificationDefinitionsSet & rhs) const;
+    bool operator!=(const ModificationDefinitionsSet& rhs) const;
     //@}
 
+    /**
+       @brief Finds modifications in the set that match a given (delta) mass
+
+       @param matches Matching modifications (output), sorted by mass error
+       @param mass Query mass
+       @param residue Query residue
+       @param term_spec Query term specificity
+       @param consider_variable Consider the variable modifications?
+       @param consider_fixed Consider the fixed modifications?
+       @param is_delta Is @p mass a delta mass (mass difference)?
+       @param tolerance Numeric tolerance for mass matching
+
+       @throw Exception::IllegalArgument if both @p consider_variable and @p consider_fixed are false
+    */
+    void findMatches(std::multimap<double, ModificationDefinition>& matches, double mass, const String& residue = "", ResidueModification::TermSpecificity term_spec = ResidueModification::NUMBER_OF_TERM_SPECIFICITY, bool consider_fixed = true, bool consider_variable = true, bool is_delta = true, double tolerance = 0.01) const;
```

Comment:
Hm, `multimap` is the obvious choice here, but an alternative would be `set<pair<double, ModificationDefinition> >`. Would that help?

Is there a good reason why Cython doesn't support `multimap`?